src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/HttpCacheProxyFactory.java

```diff
@@ -264,12 +276,43 @@ else if (headers.anyMatch(CacheDirectives.IS_ONLY_IF_CACHED))
         }
         else if (isRequestCacheable)
         {
-            newStream = newCacheableRequestStream(initial,
-                                                  routeId,
-                                                  initialId,
-                                                  resolveId,
-                                                  headers,
-                                                  requestHash);
+            counters.requestsCacheable.getAsLong();
+            HttpProxyCacheableRequestGroup group = supplyCacheableRequestGroup(requestHash);
+
+            HttpHeaderFW authorizationHeader = headers.matchFirst(h -> AUTHORIZATION.equals(h.name().asString()));
+            if (authorizationHeader != null)
+            {
+                group.authorizationHeader(authorizationHeader.value().asString());
+            }
+
+            String ifNoneMatch = getHeader(headers, IF_NONE_MATCH);
+            DefaultCacheEntry cacheEntry = defaultCache.get(requestHash);
+
+            if (hasMaxAgeZero(headers) &&
+                group.hasQueuedRequests() &&
+                (ifNoneMatch == null || ifNoneMatch.equals(group.ifNoneMatchHeader())) &&
+                cacheEntry != null &&
+                cacheEntry.canServeRequest(headers, authorizationScope))
```

Comment:
Suggest simplifying the overall approach and moving this logic into `defaultCache.matchCacheableRequest(...)` by adding a boolean to cacheEntry tracking whether or not entry is being kept up-to-date by group request (set by group request when initiated and unset when group response completed), and only in that case to also permit `max-age=0` as a cache hit.